### PR TITLE
Add Slicer to Zero Filter in ARMA (FPOG LWRS Milestone)

### PR DESCRIPTION
**Pull request by dylanjm**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)

#990 -- Please notify if a better issue for this PR exists. 

##### What are the significant changes in functionality due to this change request?

This PR modifies how a slicer is used in setting up a zero filtered ARMA. Note this change might affect #1716. This code change is required in order to run multi-target ARMAs that need zero filtering. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Review by PaulTalbot-INL — APPROVED** [review]
Changes are in alignment with standards.

**Comment by PaulTalbot-INL** [comment]
Do we have a test to cover multitarget zero-filter ARMAs?

**Comment by dylanjm** [comment]
> Do we have a test to cover multitarget zero-filter ARMAs?

It appears that `/tests/framework/ROM/TimeSeries/ARMA/zero_filter.xml` is multi-target and uses zero filter. So I suppose I'm not quite sure what is causing the error that this PR fixes.

**Comment by PaulTalbot-INL** [comment]
Yeah, I agree, I'd love to understand what this fixes, so we can test it against future failure.

**Comment by dylanjm** [comment]
> Does that test use clustering though? I think that might be the key (multitarget clustered zerofilter)

I think you're right. The current test doesn't use clustering. Should I add to this test or create another test? I can ship it with this PR.

**Comment by PaulTalbot-INL** [comment]
> Should I add to this test or create another test? I can ship it with this PR.

I'm okay with either one; I think a separate test will make it easier to find what broke on a change. I see this could quickly spiral out of control with a lot of features that can be mixed-and-matched; we should think about a testing plan for TSA when we move into clustering more heavily!

**Comment by dylanjm** [comment]
@PaulTalbot-INL I added a ZeroFilterCluster test and used data from our previous milestone. Let me know if it all checks out. Thanks.

File: tests/framework/ROM/TimeSeries/ARMA/zero_filter_cluster.xml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    </description>
+  </TestInfo>
+
+  <RunInfo>
+    <WorkingDir>ZeroFilterCluster</WorkingDir>
+    <Sequence>load, train, meta, serialize, sample</Sequence>
+  </RunInfo>
+
+  <Files>
+    <Input name="input">Data.csv</Input>
```
Comment (PaulTalbot-INL) [inline tests/framework/ROM/TimeSeries/ARMA/zero_filter_cluster.xml:19]: The 8760 training file might be considered too much for a regression test ... can we hack it down into a week with 2 clusters or something? or will that break the test too hard?

**Review by PaulTalbot-INL — APPROVED** [review]

**Pull request merged**